### PR TITLE
Initial commit of serverspec test harness

### PR DESCRIPTION
**Pull request by tayzlor**
This is a start towards #101. Opening up for initial review, with some caveats -

- Not been developed/tested yet on Digitalocean
- Not been developed/tested yet on AWS

**Outstanding questions/thinking**

- Should we run serverspecs by default, or have them off and allow them to be run by toggling some variable?
- ```tests/properties.yml``` references the hostnames > specs we need to test. Our hostnames are not consistent across providers (vagrant/do/aws) - we either need to make them consistent, or handle it otherwise
- We probably need to increase the test coverage (but this could be done afterwards)
- On vagrant the slave test fails (the slave service is not running correctly - this is a bug in the vagrant provisioning) - created #172 
- On vagrant the master test fails to see the marathon service - I believe this is because there are 2 startup scripts for marathon (one via upstart and one via init.d) and it was looking at the init.d one, which was stopped - so we might need to alter that test or use the init.d startup script - both appear to work fine (I manually started up with both scripts and they both worked)

### CODE REVIEW

File: tests/test.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+      command: gem install bundler --no-ri --no-rdoc
+      args:
+        creates: /usr/local/bin/bundler
+    - name: install bundle files
+      command: bundle install --path vendor
+      args:
+        chdir: "{{ tests_path }}"
+        creates: "{{ tests_path }}/vendor"
+    - name: run serverspec tests
+      command: "bundle exec rake serverspec:{{ ansible_hostname }}"
```
Comment (enxebre) [inline tests/test.yml]: can we just use tags instead of hostname so it would be consistent across envs

File: tests/properties.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
+master1:
```
Comment (enxebre) [inline tests/properties.yml]: If we use tags instead hostname (see comment below) we could do dry here and just have:
```
+master_tag:
+  :roles:
+    - dnsmasq
+    - mesos_master
+    - marathon
+    - zookeeper
+slave_tag:
+  :roles:
+    - dnsmasq
+    - mesos_slave
+    - weave
+    - docker
```

**Comment by wallies** [comment]
think serverspecs should be a variable and we use tags like alberto mentioned above. fixed marathon spec by running it under upstart.

**Comment by wallies** [comment]
#172 is fixed now, so is the marathon service. So just need to fix the tags

File: tests/spec/marathon/marathon_spec.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+describe package('marathon') do
+  it { should be_installed }
+end
+
+describe service('marathon') do
+  it { should be_running.under('upstart') }
+end
+
+#describe port(8080) do
```
Comment (tayzlor) [inline tests/spec/marathon/marathon_spec.rb]: This test was failing

File: .gitignore [hunk 4]
```diff
@@ -9,3 +9,6 @@ packer/build
 .wercker
 *.cache
 ssh.config
+vendor
+.bundle
+.kitchen
```
Comment (enxebre) [inline .gitignore]: Do we need .kitchen?

File: roles/serverspec/defaults/main.yml [hunk 5]
```diff
@@ -0,0 +1,5 @@
+---
+# defaults file for serverspec
+serverspec_run_tests: false
+serverspec_tests_path: /vagrant/tests
```
Comment (enxebre) [inline roles/serverspec/defaults/main.yml]: Do we need to be committed to vagrant in the naming?

File: .gitignore (hunk 4 above)
Comment (tayzlor) [inline .gitignore]: i was experimenting with it and added it, no harm to include it anyway

File: roles/serverspec/defaults/main.yml (hunk 5 above)
Comment (tayzlor) [inline roles/serverspec/defaults/main.yml]: this is just the default value - can be overridden elsewhere

**Pull request merged**